File: sdks/java/src/main/java/com/blok/blok/node/CapabilityManifest.java
package com.blok.blok.node;

import java.util.List;

/** Language-neutral v1 operational metadata from ADR 0003. */
public record CapabilityManifest(
        String version,
        String classification,
        List<String> effects,
        List<String> capabilities,
        List<String> secrets,
        String determinism,
        String idempotency,
        String maturity,
        ResourceBounds resources,
        List<String> runtimes,
        List<String> triggers) {

    public record ResourceBounds(
            Long maxDurationMs,
            Long maxMemoryBytes,
            Long maxInputBytes,
            Long maxOutputBytes,
            Long maxConcurrency) {
    }
}

File: sdks/csharp/src/Blok.Core/Node/CapabilityManifest.cs
namespace Blok.Core.Node;

/// <summary>Language-neutral v1 operational metadata from ADR 0003.</summary>
public sealed record CapabilityManifest(
    string Version,
    string Classification,
    IReadOnlyList<string> Effects,
    IReadOnlyList<string> Capabilities,
    IReadOnlyList<string> Secrets,
    string Determinism,
    string Idempotency,
    string Maturity,
    CapabilityResourceBounds? Resources = null,
    IReadOnlyList<string>? Runtimes = null,
    IReadOnlyList<string>? Triggers = null);

public sealed record CapabilityResourceBounds(
    long? MaxDurationMs = null,
    long? MaxMemoryBytes = null,
    long? MaxInputBytes = null,
    long? MaxOutputBytes = null,
    long? MaxConcurrency = null);
